### DISCUSSION TITLE
Agent in Golang?

### DISCUSSION THREAD

**Discussion by jniltinho** [glpi-project/glpi-agent#638] [Ideas]
### Your idea

Do you have plans to create this Agent in Golang?, because that way we wouldn't need a lib in Perl and we would be able to put it in more OS.
We could use go-lua, this makes it simpler to convert Perl scripts to Lua, then gradually migrate to native Golang functions.

https://github.com/Shopify/go-lua

**Comment by g-bougard** [comment]
Hi @jniltinho 

this is our plan to make evolve glpi-agent to use golang as main language. I don't think we will use go-lua to gradually migrate native to golang. Using another old language with too little community is not a good idea. In my though, we will better work first on migrating the core in golang and add support for scripts languages for modules, of course perl to reuse current modules, but probably python and maybe js. This is another way to gradually migrate to full native golang, but this may also permit contributions from people not knowing go or perl.

Actually, I have somewhere a branch where a wrote glpi-injector in go. And I don't have actually enough free time to fully work on go support. We are even looking for a skilled go developer.

**Comment by jniltinho** [comment]
Hi @g-bougard,

_Actually, I have somewhere a branch where a wrote glpi-injector in go. And I don't have actually enough free time to fully work on go support. We are even looking for a skilled go developer._

I can help with this project since I have a lot of knowledge in Perl and I'm already doing a lot of things currently in Golang, I'm having difficulty placing the agent in Perl in an environment where we can't install many libs in Perl, so with this migration everyone will be helped.
Can you give me the links so I can start studying the project, as I would like to help.
I prefer to do 100% in Golang and in the future add support for Python or even Lua as mentioned, but my focus will be 100% on Golang.

In the past I wrote functions in Golang to retrieve inventory from Linux and Windows computers using Golang, but my project was discontinued because few on the team had knowledge of Golang, this project was called Runbook, but in the end they migrated to Ruby to use in conjunction with Puppet, as Ruby was embedded in Puppet.
I think we can evolve this agent project using Golang.

**Reply by g-bougard** [reply]
Nice to hear.

I must check, but I think I still didn't published the branch anywhere. It is anyway related to one script, [glpi-injector](https://github.com/glpi-project/glpi-agent/blob/develop/bin/glpi-injector). It doesn't use  the core service but have to communicate with a glpi server to submit inventory.

I'll tell you when I'll have published the branch.

**Reply by g-bougard** [reply]
Indeed, I still published that code in develop branch here: https://github.com/glpi-project/glpi-agent/blob/develop/src/cmd/glpi-injector/main.go
